### PR TITLE
Update numerical methods

### PR DESCRIPTION
**Pull request by pstaabp**
This adds some functionality to the macro `PGnumericalmacros.pl` as well as includes a test suite for the macro.  Also, cleaned up the code and improved the documentation. 

Can provide some problems if needed.

### CODE REVIEW

**Comment by Alex-Jordan** [comment]
There's a lot to review here. Would it be hard to cut this into separate PRs? The first one could just get code cleanup out of the way without the tests or the new tools. Or something like that.

**Comment by pstaabp** [comment]
@Alex-Jordan I can do that.  Would you like separate PRs.  Or separate commits?  I could split either way to 1) code cleanup and then 2) new functionality.

File: macros/math/PGnumericalmacros.pl [hunk 1]
```diff
@@ -19,9 +19,7 @@ =head1 NAME
 
 =cut
 
-BEGIN {
-	be_strict();
-}
+BEGIN { strict->import; }
```
Comment (drgrice1) [inline macros/math/PGnumericalmacros.pl:22]: This is already on line 42 below.  One of these needs to be removed.

**Comment by pstaabp** [comment]
PR #1022 now has this without any of the new code. I will close this one soon.

**Comment by pstaabp** [comment]
closing in favor of #1022

**Pull request closed without merging**